core/src/main/java/org/elasticsearch/index/query/ScriptQueryBuilder.java

```diff
@@ -19,24 +19,50 @@
 
 package org.elasticsearch.index.query;
 
+import org.apache.lucene.index.LeafReaderContext;
+import org.apache.lucene.search.IndexSearcher;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.RandomAccessWeight;
+import org.apache.lucene.search.Weight;
+import org.apache.lucene.util.Bits;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
 import org.elasticsearch.common.xcontent.XContentBuilder;
-import org.elasticsearch.script.Script;
+import org.elasticsearch.script.*;
 import org.elasticsearch.script.Script.ScriptField;
+import org.elasticsearch.search.lookup.SearchLookup;
 
 import java.io.IOException;
+import java.util.Objects;
 
 public class ScriptQueryBuilder extends AbstractQueryBuilder<ScriptQueryBuilder> {
 
     public static final String NAME = "script";
 
-    static final ScriptQueryBuilder PROTOTYPE = new ScriptQueryBuilder((Script) null);
+    static final ScriptQueryBuilder PROTOTYPE = new ScriptQueryBuilder();
 
-    private Script script;
+    private final Script script;
+    
+    private ScriptQueryBuilder() {
+        this.script = null;
+    }
 
     public ScriptQueryBuilder(Script script) {
+        if (script == null) {
+            throw new IllegalArgumentException("supplied script cannot be null.");
+        }
```

Comment:
I am not sure what to know you mean by do the same as in other queries. Can you point me to a particular query? Maybe I'm missing something, but it seems we had settled for not checking for null in the constructor, and only do it in validate. Checking for null in the constructor would make the tests fail because of our prototype pattern, unless we use empty constructors.